src/collections/cursor.ts

```diff
@@ -135,7 +138,12 @@ export class FindCursor {
         if (this.nextPageState == null) {
             this.exhausted = true;
         }
-        this.page = Object.keys(resp.data.documents).map(i => resp.data.documents[i]);
+        this.page = resp.data.documents;
+        if (options != null && options.includeSortVector) {
+            this.page.forEach(doc => {
+                doc.$sortVector = resp.status.sortVector;
```

Comment:
I like the idea of returning the sortVector as part of the doc. It looks good for the findOne.

For the find Call, should we consider updating only the first doc in the response, instead of adding the same vector on each response document, so that the payload size will be drastically reduced (thinking of a case as an example where a sort vector that has 1500 dimensions is returned as part of each doc which is redundant)?

Reply:
For the most part, duplicated $sortVector in result payload isn't much of a concern. However, there are cases where it is, so I think you have a point.

 Unless you specify $sortVector as a field or virtual in your schema, $sortVector won't end up in JSON.stringify() output. And mongoose lets you configure whether virtuals end up in JSON.stringify output.

However, if you use lean(), $sortVector will be a property on all documents in JSON.stringify() output.

What do you think about just setting $sortVector as a property on the result array? That way we also get $sortVector even if there's no results

Reply:
Understood on the stringify part, thanks!
I'm sorry, I'm not getting the proposed alternate solution. Is the idea here to return `$sortVector` as the first element in the result array?

Reply:
@kathirsvn in 4950005 I created an alternative approach where, with `find()`, you need to use a query cursor to get the sort vector. So `await cursor.getSortVector()` instead of decorating documents with `$sortVector`. This approach has a couple of benefits:

1. No duplicated $sortVector when serializing lean documents
2. Still can get sort vector even if there's no results
3. No need for an extra document property

What do you think?

Reply:
This looks good to me as this avoids duplicates primarily.